crypto/armcap.c

```diff
@@ -168,42 +174,61 @@ void OPENSSL_cpuid_setup(void)
             if (hwcap & HWCAP_CE_SHA256)
                 OPENSSL_armcap_P |= ARMV8_SHA256;
 
-# ifdef __aarch64__
+#  ifdef __aarch64__
             if (hwcap & HWCAP_CE_SHA512)
                 OPENSSL_armcap_P |= ARMV8_SHA512;
-# endif
+#  endif
         }
     } else if (sigsetjmp(ill_jmp, 1) == 0) {
-        _armv7_neon_probe();
-        OPENSSL_armcap_P |= ARMV7_NEON;
-        if (sigsetjmp(ill_jmp, 1) == 0) {
-            _armv8_pmull_probe();
-            OPENSSL_armcap_P |= ARMV8_PMULL | ARMV8_AES;
-        } else if (sigsetjmp(ill_jmp, 1) == 0) {
-            _armv8_aes_probe();
-            OPENSSL_armcap_P |= ARMV8_AES;
-        }
-        if (sigsetjmp(ill_jmp, 1) == 0) {
-            _armv8_sha1_probe();
-            OPENSSL_armcap_P |= ARMV8_SHA1;
-        }
-        if (sigsetjmp(ill_jmp, 1) == 0) {
-            _armv8_sha256_probe();
-            OPENSSL_armcap_P |= ARMV8_SHA256;
+        if (mask & ARMV7_NEON) {
+            _armv7_neon_probe();
+            OPENSSL_armcap_P |= ARMV7_NEON;
+            if (mask & ARMV8_PMULL) {
+                if (sigsetjmp(ill_jmp, 1) == 0) {
+                    _armv8_pmull_probe();
+                    OPENSSL_armcap_P |= ARMV8_PMULL | ARMV8_AES;
+                    goto test_sha1;
+                }
+            }
+            if (mask & ARMV8_AES) {
```

Comment:
??? What's wrong with else here?

Reply:
if mask =  PMULL | AES  but runtime check fails the AES check will not be tried.

Reply:
previously the else was on the "if (sigsetjmp...)"
but now it is on the "if (mask & PMULL)"
either way I can miss the AES check.